src/components/victory-candlestick/helper-methods.js

```diff
@@ -0,0 +1,156 @@
+    const calculatedValues = this.getCalculatedValues(props, defaultStyles);
+    const { data, style, scale } = calculatedValues;
+    return data.reduce((memo, datum, index) => {
+      const eventKey = datum.eventKey;
+      const x = scale.x(datum.x);
+      const y1 = scale.y(datum.high);
+      const y2 = scale.y(datum.low);
+      const candleHeight = Math.abs(scale.y(datum.open) - scale.y(datum.close));
+      const y = scale.y(Math.max(datum.open, datum.close));
+      const size = this.getSize(datum, props, calculatedValues);
+      const dataStyle = Object.assign(this.getDataStyles(datum, style.data, props));
+      const dataProps = {
+        x, y, y1, y2, candleHeight, size, scale, data, datum,
+        index, style: dataStyle, padding: props.padding, width: props.width
+      };
+
+      const text = this.getLabelText(props, datum, index);
+      const labelStyle = this.getLabelStyle(style.labels, dataProps);
+      const labelProps = {
+        style: labelStyle,
+        x,
+        y: y - labelStyle.padding,
+        text,
+        index,
+        scale,
+        datum: dataProps.datum,
+        textAnchor: labelStyle.textAnchor,
+        verticalAnchor: labelStyle.verticalAnchor || "end",
+        angle: labelStyle.angle
+      };
+      memo[eventKey] = {
+        data: dataProps,
+        labels: labelProps
+      };
+      return memo;
+    }, {});
+  },
+
+  getCalculatedValues(props, defaultStyles) {
+    const style = Helpers.getStyles(props.style, defaultStyles, "auto", "100%");
+    const data = Events.addEventKeys(props, this.getData(props));
+    // console.log(this.getData({data: [{x: 5, open: 10, close: 20, high: 25, low: 5}]}));
+    const range = {
+      x: Helpers.getRange(props, "x"),
+      y: Helpers.getRange(props, "y")
+    };
+    const domain = {
+      x: this.getDomain(props, "x"),
+      y: this.getDomain(props, "y")
+    };
+    const scale = {
+      x: Scale.getBaseScale(props, "x").domain(domain.x).range(range.x),
+      y: Scale.getBaseScale(props, "y").domain(domain.y).range(range.y)
+    };
+    return {data, scale, style};
+  },
+
+  getData(props) {
+    const data = props.data ? props.data : this.generateData(props);
+    return data.map((datum) => {
+      const x = datum.x;
+      const y = [datum.open, datum.close, datum.high, datum.low];
+      return Object.assign(
+        {},
+        datum,
+        {x, y}
+        );
+    });
+  },
+
+  generateData(props) {
+    // create an array of values evenly spaced across the x domain that include domain min/max
+    const domain = props.domain ? (props.domain.x || props.domain) :
+      Scale.getBaseScale(props, "x").domain();
+    const samples = props.samples || 1;
+    const max = Math.max(...domain);
+    const values = Array(...Array(samples)).map((val, index) => {
+      const v = (max / samples) * index + Math.min(...domain);
+      return { x: v, open: v, close: v + 2, low: v - 3, high: v + 5 };
```

Comment:
Oh I see it now. This is your default data. I don't like having this hard coded into a method. I'm okay with the default data existing as a defaultProp as it does in `VictoryBar`. That way it will get documented in the props table.